File: fibonacci/fib.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

// fib returns a function that returns
// successive Fibonacci numbers.
func fib(n int64) int64 {
	if n < 2 {
		return int64(n)
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	n, _ := strconv.Atoi(os.Args[1])

	fmt.Printf("LANGUAGE GO: %d\n", fib(int64(n)))
}

File: fibonacci/fib.rs
use std::*;
// fib returns a function that returns
// successive Fibonacci numbers.
fn fib(n: u64) -> u64 {
	if n < 2 {
		return n;
	}
	return fib(n-1) + fib(n-2);
}

fn main() {
    let args: Vec<_> = env::args().collect();
    let n: u64 = args[1].parse::<u64>().unwrap();
    let result = fib(n);
	println!("LANGUAGE Rust: {0}", result);
}
